[14.0][MIG] base_import_security_group

StefanRijnhart opened the pull request:
Continued from https://github.com/OCA/server-ux/pull/342 by @Chandresh-SerpentCS

Additional changes:

* Patch existing class instead of replacing it, for better compatibility with other modules inheriting from the same class.
* To circumvent problems executing async calls in a non-widget class, pass the user's import permission through the web session.
* Adapt tests to consult he `Favorites` dropdown in the 14.0 GUI.
* Rebased according to OCA conventions.
* In the tour, don't target the `Users` menu which is the default menu of the `Settings` menu when no other module is installed, and in that case clicking the menu in the tour is a void action that makes the tests fail. Target the `Companies` menu item instead.

espo-tony committed:
[10.0][ADD] Module base_import_csv_optional

This module makes importing data from CSV files optional for each user, allowing it only for those users belonging to a specific group.

[FIX] Solved flake8 error E305: expected 2 blank lines after class or function definition, found 1

[FIX] Renamed base_import_csv_optional in base_import_security_group

[FIX] Solved flake8 error E501: line too long;

[FIX] Removed unnecessary monkeypatch and extended Base.load method;

[FIX] Bugfixing in Base.load method;

[FIX] Code refactoring

[IMP] Tests created

StefanRijnhart committed:
[ADD] js tests

espo-tony committed:
[FIX] Fixed Phantom JS tests

[FIX] Updated travis postgresql version

[FIX] Updated phantom_js version in travis

[FIX] Configured travis to run tests for this module in isolation mode

[FIX] Configured travis to run tests for this module in isolation mode

[FIX] Added group to demo_user for test purposes

[FIX] Changed waiting condition on Phantom JS test

[FIX] Removed Travis configuration for isolated test environment

[FIX] Changed waiting condition in phantom_js test

[FIX] Changed target action in phantom_js test

[FIX] Changed target action in phantom_js test

espo-tony committed:
[FIX] Updated module informations

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

[UPD] Update base_import_security_group.pot

[UPD] Update base_import_security_group.pot

[UPD] README.rst

Commit:
[11.0][MIG] base_import_security_group: adapt to v11

ajayjaviya committed:
[MIG] base_import_security_group: Migration to 12.0

oca-travis committed:
[UPD] Update base_import_security_group.pot

OCA-git-bot committed:
[UPD] README.rst

StefanRijnhart committed:
[FIX] base_import_security_group browser test

StefanRijnhart committed:
[FIX] Give travis some more time to determine if user is allowed to import

victoralmau committed:
[IMP] base_import_security_group: black, isort, prettier

victoralmau committed:
[MIG] base_import_security_group: Migration to 13.0

oca-travis committed:
[UPD] Update base_import_security_group.pot

OCA-git-bot committed:
[UPD] README.rst

Chandresh-OSI committed:
[MIG] base_import_security_group: Migration to 14.0

StefanRijnhart committed:
[IMP] base_import_security_group: black, isort, prettier

StefanRijnhart commented:
~~Darn. I missed that Odoo skips the javascript tests, which fail here. Having a look.~~ Tests were not adapted to the situation in 14.0. To further improve inheratibility of the `ImportMenu` class I refactored the javascript so that it is patched instead of replaced by this module.

PieterPaulussen commented on base_import_security_group/static/src/js/import.js:
```diff
@@ -0,0 +1,24 @@
+    ImportMenu.shouldBeDisplayed = function (env) {
+        return (
+            shouldBeDisplayed_orig(env) &&
+            this.env.session.base_import_security_group__allow_import === 1
```
also valid apparently: `env.session...` instead of `this.env`
But I do not really know what difference it would make.

StefanRijnhart replied:
You're right! `this.env` seems to be a given in owl classes but if we get handed an `env` we better use that one.

StefanRijnhart committed:
[RFR] base_import_security_group: pass ability through the web session

* Patch existing class instead of duplicating it, for better compatibility
  with other modules enhancing the same class.
* To circumvent problems executing async calls in a non-widget class,
  pass the user's import permission through the web session.
* Adapt tests to consult he Favorite dropdown in the 14.0 GUI.
* In the tour, don't target the Users menu which is the default menu of the
  Settings menu when no other module is installed, and in that case clicking
  the menu in the tour is a void action that makes the tests fail. Target the
  Companies menu item instead.

dsolanki-initos reviewed (approved):
LGTM. Code review

dsolanki-initos commented:
It would be great if the maintainers could merge it

dreispt commented:
/ocabot merge nobump
